build.rs

```diff
@@ -485,37 +484,31 @@ fn build_c_code(
     );
 }
 
+fn new_build(target: &Target, include_dir: &Path) -> cc::Build {
+    let mut b = cc::Build::new();
+    configure_cc(&mut b, &target, include_dir);
+    b
+}
+
 fn build_library(
     target: &Target,
     out_dir: &Path,
     lib_name: &str,
     srcs: &[PathBuf],
     additional_srcs: &[PathBuf],
 ) {
+    let mut c = new_build(&target, out_dir);
+
     // Compile all the (dirty) source files into object files.
     let objs = additional_srcs
         .iter()
         .chain(srcs.iter())
-        .map(|f| compile(f, target, out_dir, out_dir))
+        .map(|f| compile(&c, f, target, out_dir, out_dir))
```

Comment:
Now all the calls to `compile()` share a single build configuration that is constructed once per library, instead of once per source file.